validator/project_validator.go

```diff
@@ -222,58 +222,50 @@ func getDistrosForProject(ctx context.Context, projectID string) (ids []string,
 	return ids, aliases, nil
 }
 
-func CheckProject(ctx context.Context, project *model.Project, config *model.ProjectConfig, ref *model.ProjectRef, settings *evergreen.Settings, includeLong, projectRefProvided, projectRefErrored bool) ValidationErrors {
+// CheckProject calls the validating logic for a Project's configuration.
+// That is, ProjectErrors, ProjectWarnings, ProjectConfigErrors,
+// ProjectSettings, and AliasWarnings. If a respective item is nil,
+// it will not check it (e.g. if the config is nil, it does not check the config).
+// projectRefId is used to determine if there is a project specified and
+// projectRefErr is used to determine if there was a problem retrieving
+// the ref, both output different warnings for the project.
+func CheckProject(ctx context.Context, project *model.Project, config *model.ProjectConfig, ref *model.ProjectRef, includeLong bool, projectRefId string, projectRefErr error) ValidationErrors {
 	isConfigDefined := config != nil
 	verrs := CheckProjectErrors(ctx, project, includeLong)
-	if !projectRefProvided {
+	verrs = append(verrs, CheckProjectWarnings(project)...)
+	if config != nil {
+		verrs = append(verrs, CheckProjectConfigErrors(config)...)
+	}
+
+	if projectRefId == "" {
 		verrs = append(verrs, ValidationError{
-			Message: "no project specified; validation will proceed without checking project settings",
+			Message: "no project specified; validation will proceed without checking project settings and alias coverage",
```

Comment:
nit: maybe we should change this wording to "will not check ..." instead of "will proceed", since the latter wording relies on ordering (like, we're doing this check last now so there's no other validation to proceed to)